Client/Circle.cpp

```diff
@@ -1,25 +1,32 @@
 #include "stdafx.h"
 #include "Circle.h"
 
-Circle::Circle() : Shape(), radius(0) {}
+Circle::Circle() :Shape(), radius(0) { setCenter(Vector2D()); }
 
-Circle::Circle(const Vector2D& v, const unsigned int r, const std::shared_ptr<const Color>& c) : Shape() {
-	vertices.push_back(v);
-	color = c;
-	radius = r;
-}
+Circle::Circle(const Vector2D& v) : Circle() { setCenter(v); }
 
-Circle::Circle(const Circle& c) {
-	vertices = c.getVertices();
-	color = c.color;
-	radius = c.getRadius();
-}
+Circle::Circle(const Vector2D& v, const unsigned int r, const std::shared_ptr<const Color>& c) : Shape(v,c), radius(r) {}
 
-Circle::~Circle() {
-}
+Circle::Circle(const Circle& c) : Shape(c), radius(c.getRadius()){}
+
+Circle::~Circle() {}
 
 const int Circle::getRadius() const { return radius; }
 
 void Circle::setRadius(const int i) { radius = i; }
 
-string Circle::getName() const { return string("circle"); }
\ No newline at end of file
+void Circle::setCenter(const Vector2D& v) {
+	if (vertices.empty()) vertices.push_back(v);
+	else vertices[0] = v;
+}
```

Comment:
If the vector is initialized in the constructor, you don't need to make the test here.